{"error": null, "issues": [{"address": 2425, "code": "amt),\n            uint64(now)\n        );\n\n        ", "debug": "", "description": "This contract executes a message call to an address provided as a function argument. Generally, it is not recommended to call user-supplied addresses using Solidity's call() construct. Note that attackers might leverage reentrancy attacks to exploit race conditions or manipulate this contract's state.", "filename": "/temp/Oasis_Trade-0.sol", "function": "cancel(uint256)", "lineno": 333, "title": "Message call to external contract", "type": "Warning"}, {"address": 3055, "code": "n offers[id].o", "debug": "", "description": "The contract account state is changed after an external call. Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.", "filename": "/temp/Oasis_Trade-0.sol", "function": "cancel(uint256)", "lineno": 237, "title": "State change after external call", "type": "Warning"}, {"address": 4888, "code": "nd > offer.buy_amt)\n        {\n            re", "debug": "The exception is triggered under the following conditions:\n\nstorage_keccac_UDiv(calldata_SimpleMarket_4,_1) + 2: 0x200100000000000000000000000000000\ncalldata_SimpleMarket_32 + 4: 0x40000eef77733356543c4d0d8682bada\nstorage_2: 0x0\nstorage_4 + keccac_UDiv(calldata_SimpleMarket_4,_1): 0xffffffffffffffff00\ncalldata_SimpleMarket_0: 0x4960645500000000000000000000000000000000000000000000000000000000\ncalldatasize_SimpleMarket: 0x4\nstorage_keccac_UDiv(calldata_SimpleMarket_4,_1) + 0: 0x0\ncallvalue: 0x0\n", "description": "A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. ", "filename": "/temp/Oasis_Trade-0.sol", "function": "take(bytes32,uint128)", "lineno": 281, "title": "Exception state", "type": "Informational"}, {"address": 7643, "code": "\n        public\n    {\n        require(buy(uint2", "debug": "", "description": "This contract executes a message call to an address provided as a function argument. Generally, it is not recommended to call user-supplied addresses using Solidity's call() construct. Note that attackers might leverage reentrancy attacks to exploit race conditions or manipulate this contract's state.", "filename": "/temp/Oasis_Trade-0.sol", "function": "_function_0x093f5198", "lineno": 398, "title": "Message call to external contract", "type": "Warning"}, {"address": 8241, "code": "n offers[id].o", "debug": "", "description": "The contract account state is changed after an external call. Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.", "filename": "/temp/Oasis_Trade-0.sol", "function": "_function_0x093f5198", "lineno": 237, "title": "State change after external call", "type": "Warning"}, {"address": 8341, "code": "losed());\n     ", "debug": "storage_0: 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff\ncalldata_SimpleMarket_32 + 4: 0x2010000000000000000000000000000\ncalldata_SimpleMarket_32 + 68: 0xffffffffffffffffffffffffffffffff\ncalldata_SimpleMarket_4: 0x2000000000000000000000000000000\ncalldata_SimpleMarket_32 + 36: 0xffffffffffffffffffffffffffffffff\nstorage_2: 0x0\ncalldata_SimpleMarket_0: 0x93f519800000000000000000000000000000000000000000000000000000000\ncalldatasize_SimpleMarket: 0x4\ncallvalue: 0x0\n", "description": "A possible integer overflow exists in the function `_function_0x093f5198`.\nThe addition or multiplication may result in a value higher than the maximum representable integer.", "filename": "/temp/Oasis_Trade-0.sol", "function": "_function_0x093f5198", "lineno": 421, "title": "Integer Overflow ", "type": "Warning"}], "success": true}
